packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -4186,6 +4285,17 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
     }
   }
 
+  void _updateBuildScopeRecursively() {
+    if (buildScope == _parent?.buildScope) {
+      return;
+    }
+    _inDirtyList = false;
```

Comment:
This flag is used by `scheduleBuildFor` to determine whether the node needs to be added to the dirty list. After switching to a different build scope we have to unset the flag so it can be add to the dirty list of the new build scope. `scheduleBuildFor` will be called by `Element.activate`.

Removing from list is too slow so `_flushDirtyElements` just skips the elements that have been reparented to a different build scope.

Reply:
> After switching to a different build scope we have to unset the flag so it can be add to the dirty list of the new build scope. scheduleBuildFor will be called by Element.activate.

Maybe leave this explanation as a comment in the code?